tests/test_xjoin.py

```diff
@@ -62,6 +62,28 @@ def test_host_request_xjoin_status_403(patch_xjoin_post, api_get):
     assert response_status == 500
 
 
+def test_host_request_xjoin_invalid_response_none(patch_xjoin_post, api_get):
+    patch_xjoin_post(response={"data": None}, status=200)
+    request_id = generate_uuid()
+
+    response_status, response_data = api_get(
+        HOST_URL, extra_headers={"x-rh-insights-request-id": request_id, "foo": "bar"}
+    )
+
+    assert response_status == 503
+
+
+def test_host_request_xjoin_invalid_response_empty(patch_xjoin_post, api_get):
+    patch_xjoin_post(response={"data": {}}, status=200)
+    request_id = generate_uuid()
+
+    response_status, response_data = api_get(
+        HOST_URL, extra_headers={"x-rh-insights-request-id": request_id, "foo": "bar"}
+    )
+
+    assert response_status == 503
+
+
```

Comment:
Looks good, but this can also be parameterized to reduce duplication:
```suggestion
@pytest.mark.parametrize("response_data", (None, {}))
def test_host_request_xjoin_invalid_response_none(patch_xjoin_post, api_get, response_data):
    patch_xjoin_post(response={"data": response_data}, status=200)
    request_id = generate_uuid()

    response_status, response_data = api_get(
        HOST_URL, extra_headers={"x-rh-insights-request-id": request_id, "foo": "bar"}
    )

    assert response_status == 503


```

Reply:
Done.